### ISSUE TITLE
go get fails to import cri-resource-manager

### ISSUE THREAD

**Issue by bart0sh** [intel/cri-resource-manager#152]
Here is how it happens:
```
$ go get github.com/intel/cri-resource-manager/pkg/sysfs
go: finding github.com/intel/cri-resource-manager v0.2.0
go: finding github.com/intel/cri-resource-manager/pkg latest
go: finding github.com/intel/cri-resource-manager/pkg/sysfs latest
go: finding github.com/intel/cri-resource-manager v0.2.0
go: downloading github.com/intel/cri-resource-manager v0.2.0
verifying github.com/intel/cri-resource-manager@v0.2.0: github.com/intel/cri-resource-manager@v0.2.0: reading https://sum.golang.org/lookup/github.com/intel/cri-resource-manager@v0.2.0: 410 Gone
```

The actual error is this:
```
not found: unzip /tmp/gopath/pkg/mod/cache/download/github.com/intel/cri-resource-manager/@v/v0.2.0.zip: malformed file path "pkg/sysfs/testdata/sys/devices/pci0000:00/0000:00:02.0/class": invalid char ':'
```

I'll try to come up with a fix soon.

**Comment by klihub** [comment]
Hmm, strange. Looks like the toolchain cannot deal with files with a colon in their name.

@bart0sh Since the problem seems to be only with sysfs test data, where colons do appear in file names, one ugly workaround would be to hide those files from the toolchain during go get, for instance movin them to an in-repo tarball which is then extracted  prior to running the tests (by appropriately crafted Makefile rules).

**Comment by klihub** [comment]
This is a toolchain problem. See any of the many bug reports, for instance [this](https://github.com/golang/go/issues/27642).

The suggested semi-official upstream workaround is to scatter around empty go.mod files as needed in your repository to fool the toolchain to consider the subtrees with offending file names separate packages so they get excluded from the zip created during go get.

**Comment by bart0sh** [comment]
I'm going to fix it by removing pkg/sysfs/testdata and generating test data dynamically in the test code. apart from fixing this particular issue it would also improve synchronization between test data and code.

**Comment by klihub** [comment]
> I'm going to fix it by removing pkg/sysfs/testdata and generating test data dynamically in the test code. apart from fixing this particular issue it would also improve synchronization between test data and code.

Hmm... I'm not sure if that is the right fix. I think it would be good to keep real world HW sysfs-dumps in the repo as test data and verify that our discovery code works correctly against those.

**Comment by bart0sh** [comment]
> Hmm... I'm not sure if that is the right fix

I did it already, please review. Meanwhile I'll try to follow your suggestion in another PR.

**Comment by marquiz** [comment]
Can this be closed?

**Comment by klihub** [comment]
Need to rebase and resolve conflicts.Then we'd need to merge it if it is fine with everybody, as @bart0sh needs it, IIUC,

**Comment by bart0sh** [comment]
> Can this be closed?

not until the issue is resolved I guess. It's still not resolved:
```
$ go get github.com/intel/cri-resource-manager/pkg/sysfs
go: downloading github.com/intel/cri-resource-manager v0.2.0
verifying github.com/intel/cri-resource-manager@v0.2.0: github.com/intel/cri-resource-manager@v0.2.0: reading https://sum.golang.org/lookup/github.com/intel/cri-resource-manager@v0.2.0: 410 Gone
```

**Comment by klihub** [comment]
We need to rebase+resolve conflicts+merge #158 for this.

**Comment by bart0sh** [comment]
and probably tag as go get downloads latest available tag if nothing else specified.

**Comment by ipuustin** [comment]
I tried this now and it worked for me.
